core/src/main/java/com/netflix/conductor/core/utils/SemaphoreUtil.java

```diff
@@ -62,4 +63,23 @@ public int availableSlots() {
         LOGGER.debug("Number of available permits: {}", available);
         return available;
     }
+
+    /**
+     * Signals if processing is allowed based on whether specified number of permits can be acquired.
+     *
+     * @param numSlots the number of permits to acquire
+     * @return {@code true} - if permit is acquired
+     * {@code false} - if permit could not be acquired
+     */
+    public boolean acquireSlots(int numSlots) {
```

Comment:
Does this need to be a separate method? This could be rolled into the methods defined above where numSlots=1.

Reply:
Thanks for the suggestion. Using the extended method with numSlots=1 now.